src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1943,6 +1944,53 @@ internal void IncrementPasteOffset()
             this.currentPasteOffset = (this.currentPasteOffset + PasteOffsetStep) % PasteOffsetMax;
         }
 
+        #region [Nodes Info]
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings
+        /// </summary>
+        internal bool HasWarnings
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Warning || n.State == ElementState.PersistentWarning); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings with no Geometry
+        /// </summary>
+        internal bool HasWarningsWithNoGeometry
+        {
+            get { return Nodes.Any(n => (n.State == ElementState.Warning || n.State == ElementState.PersistentWarning) && !n.Category.StartsWith("Geometry.")); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if workspace run with errors
+        /// </summary>
+        internal bool HasErrors
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Error); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if home workspace is displayed with infos
+        /// </summary>
+        internal bool HasInfos
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Info); }
+        }
+
+        /// <summary>
+        /// Indicates if the workspace is valid for sending to the FDX
+        /// </summary>
+        internal bool IsValidForFDX
+        {
+            get
+            {
+                return !HasErrors && !HasWarningsWithNoGeometry; //Pending dealing with the Gate Nodes
```

Comment:
Dont we need to check if the graph has run? Warning could be missing if the graph is just opened, right?

Reply:
But I guess we cant force user to run their graphs, did we discuss this case with @varvaratou ?

Reply:
I have moved the validation after running a graph in the Homeworkspace (The Custom node Workspace is not runnable) and also when the graph is saving only for a Homeworkspace. I don't think we want validate a graph from a Custom node Workspace, this is a question for the Business I guess.

Yes, We can dismiss a warning from a node(we are checking each node to the current validation) but we remains the dismissed warnings too through the  [Run completed with warnings] message.

I will contact Varvara about these inquiries.